README.md

```diff
@@ -1,4 +1,1060 @@
+
+Because these are system-independent outputs, you do not have access to the de-systemized `inputs` nor to `pkgs`.
+
+Only in this file is it appropriate, if needed, to reach for `nixpkgs` like so:
+```nix
+pkgs = inputs'.nixpkgs.legacyPackages.x86_64-linux;
+lib = inputs'.nixpkgs.lib;
+```
+
+The values defined in this file will be available via `self` anywhere else in your Nix code using `inputs` and `inputs'` equivalently.
+
+For example:
+```nix
+{ inputs, inputs', ... }:
+
+let 
+  x = inputs.self.lib.f inputs'.self.networks.prod; 
+in 
+  { };
+``` 
+
+### 3.6.1. `inputs'`
+
+See [`inputs`](#332-inputs) from [`haskell-project.nix`](#33-nixhaskell-projectnix).
+
+## 3.7. `nix/read-the-docs.nix`
+
+TODO 
+
+## 3.8. `nix/pre-commit-check.nix`
+
+```nix
+# Default nix/pre-commit-check.nix
+{
+  cabal-fmt.enable = false;
+  cabal-fmt.extraOptions = "";
+
+  stylish-haskell.enable = false;
+  stylish-haskell.extraOptions = "";
+
+  shellcheck.enable = false;
+  shellcheck.extraOptions = "";
+
+  prettier.enable = false;
+  prettier.extraOptions = "";
+
+  editorconfig-checker.enable = false;
+  editorconfig-checker.extraOptions = "";
+
+  nixpkgs-fmt.enable = false;
+  nixpkgs-fmt.extraOptions = "";
+
+  png-optimization.enable = false;
+  png-optimization.extraOptions = "";
+
+  fourmolu.enable = false;
+  fourmolu.extraOptions = "";
+
+  hlint.enable = false;
+  hlint.extraOptions = "";
+
+  hindent.enable = false;
+  hindent.extraOptions = "";
+}
+```
+
+Configuration for code formatters and linters.
+
+These are fed to [`pre-commit-hooks`](https://github.com/cachix/pre-commit-hooks.nix), which is run whenever you `git commit`.
+
+The `pre-commit` executable is also available in the shell.
+
+Currently 10 tools are available, and they are all disabled by default.
+
+If this file is missing, then no hooks will be run.
+
+### 3.8.1. `enable` 
+
+It is sufficient to set the `enable` flag to `true` to make the tool active.
+
+When enabled, some tools expect to find a configuration file in the root of the repository:
+
+| Tool Name | Config File | 
+| --------- | ----------- |
+| `stylish-haskell` | `.stylish-haskell.yaml` |
+| `editorconfig-checker` | `.editorconfig` |
+| `fourmolu` | `fourmolu.yaml` (note the missing dot `.`) |
+| `hlint` | `.hlint.yaml` |
+| `hindent` | `.hindent.yaml` |
+
+Currently there is no way to change the location of the configuration files.
+
+Each tool knows which file extensions to look for, which files to ignore, and how to modify the files in-place.
+
+### 3.8.2. `extraOptions` 
+
+You can *append* additional options to a tool's command by setting the `extraOptions` field.
+
+For example:
+
+```nix
+{
+  cabal-fmt.enable = true;
+  cabal-fmt.extraOptions = "--no-tabular";
+
+  fourmolu.enable = false;
+  fourmolu.extraOptions = "-o -XTypeApplications -o XScopedTypeVariables";
+}
+```
+
+## 3.9. `nix/hydra-jobs.nix`
```

Comment:
Probably future work, but might be nice to have Hydra jobs that aren't part of `required`